### PR TITLE
Add initial type definition type

### PR DESCRIPTION
**Pull request by nettofarah**
TODO:

- [x] Add base definition for CSS props
- [ ] Add definitions for enhancers
- [ ] Add definitions for helper functions

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// Definitions by: Netto Farah <https://github.com/nettofarah>
+
+import { Component, ComponentClass } from "react"
+import CSS from "csstype"
+
+
+/** Placeholder type for UI box props */
+type UIBoxProp = string | number | boolean
+
+type UIBoxIsProp = string
```
Comment (mshwery) [inline index.d.ts]: This can be a component too, right?

Comment (nettofarah) [reply index.d.ts]: Fixed!

**Comment by mshwery** [comment]
BTW I wonder if we should take the second approach recommended in the handbook: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

>There are two main ways you can publish your declaration files to npm:
>
>1. bundling with your npm package, or
>2. publishing to the [@types organization](https://www.npmjs.com/~types) on npm.
>
>If your package is written in TypeScript then the first approach is favored. Use the `--declaration` flag to generate declaration files. This way, your declarations and JavaScript always be in sync.
>
>**If your package is not written in TypeScript then the second is the preferred approach.**

(emphasis mine)

**Comment by nettofarah** [comment]
I'm down with that!
I'll close this PR, keep working on the definitions, and push it up t definitely typed.

**Pull request closed without merging**